pjsip/src/pjsip/sip_parser.c

```diff
@@ -1778,14 +1778,39 @@ static void parse_generic_array_hdr( pjsip_generic_array_hdr *hdr,
         return;
     }
 
-    pj_scan_get( scanner, &pconst.pjsip_NOT_COMMA_OR_NEWLINE, 
-                 &hdr->values[hdr->count]);
-    hdr->count++;
+    /* Skip leading comma if present (handle buggy implementations) */
+    if (*scanner->curptr == ',') {
+        pj_scan_get_char(scanner);
+        pj_scan_skip_whitespace(scanner);
+        /* If we're now at EOF or newline, we're done */
+        if (pj_scan_is_eof(scanner) || 
+            *scanner->curptr == '\r' || *scanner->curptr == '\n') 
+        {
+            goto end;
+        }
+    }
+
+    /* Only parse element if we're not at a comma or newline */
+    if (*scanner->curptr != ',' && 
+        *scanner->curptr != '\r' && *scanner->curptr != '\n') 
+    {
+        pj_scan_get( scanner, &pconst.pjsip_NOT_COMMA_OR_NEWLINE, 
+                     &hdr->values[hdr->count]);
+        hdr->count++;
+    }
```

Comment:
If possible, to avoid redundancy, remove this block and modify the loop block below to be able to skip any commas in the beginning, the middle, or the end, and only add the parsed element if it is not empty.